backend/src/services/referral.ts

```diff
@@ -33,11 +36,39 @@ export async function editReferral(
   assignedReferringStaffId: string,
   status: string,
 ) {
+  const Ref = await ReferralModel.findById(id);
+  const updateHL = Ref?.assignedHousingLocator?.toString() !== assignedHousingLocatorId;
+  const setLeased =
+    status === "leased" &&
+    Ref?.status !== "Leased" &&
+    Ref?.status !== "Denied" &&
+    Ref?.status !== "Canceled";
   await ReferralModel.findByIdAndUpdate(id, {
     assignedHousingLocator: assignedHousingLocatorId,
     assignedReferringStaff: assignedReferringStaffId,
     status,
   });
   const referral = await ReferralModel.findById(id);
+  const HL = await getUserByID(referral?.assignedHousingLocator?.toString() ?? "");
+
+  if (updateHL) {
+    if (HL !== null) {
+      await sendEmail(
+        HL.email,
+        "Referral Update",
+        `You have been assigned a new referral. Please login to the portal to view the update.`,
+      );
+    }
+  }
+  if (setLeased) {
+    if (HL !== null) {
+      await sendEmail(
+        HL?.email,
+        "Referral Update",
+        `Your referral has been marked as leased. Please login to the portal to view the update.`,
+      );
+    }
+  }
```

Comment:
> - When referral status(es) for unit is set to leased → notify all HLs associated with the ACTIVE referrals for the unit
>     - inactive referral statuses = [ “Denied”, “Canceled” ]

Oops maybe I wasn't too clear about this. When a referral gets set to "Leased", the unit for that referral is leased, so we want to notify all _other_ referrals for that unit that the unit is not available anymore. This currently looks like it just notifies the HL for the current referral.